### PR TITLE
Add C implementation for `@stdlib/math/base/assert/is-probability`

### LINKED ISSUE #761: [RFC]: Add C implementation for `@stdlib/math/base/assert/is-probability`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/assert/is-probability`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/assert/is-probability).

```c
bool stdlib_base_is_probability( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #761.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/assert/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/assert/is-probability/src/addon.c [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		return NULL;
+	}
+
+	double x;
+	status = napi_get_value_double( env, argv[ 0 ], &x );
+	assert( status == napi_ok );
+
+	bool result = stdlib_base_is_probability( x );
+    
+    int32_t out = (int32_t) result;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/src/addon.c]:
```suggestion
	int32_t out = (int32_t) result;
```

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can be reviewed.

File: lib/node_modules/@stdlib/math/base/assert/is-probability/README.md [hunk 2]
```diff
@@ -74,6 +74,97 @@ for ( i = 0; i < 100; i++ ) {
...
+
+### Usage
+
+```c
+#include "stdlib/math/base/assert/is_probability.h"
+```
+
+#### stdlib_base_is_probability( x )
+
+Tests if a `numeric` value is a probability.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/README.md]:
```suggestion
Tests if a numeric value is a probability.
```

File: lib/node_modules/@stdlib/math/base/assert/is-probability/README.md [hunk 3]
```diff
@@ -74,6 +74,97 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    bool v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( ( double ) rand() / ( double ) RAND_MAX ) * 2.0 ) - 1.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/README.md]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 2.0 ) - 1.0;
```

File: lib/node_modules/@stdlib/math/base/assert/is-probability/README.md [hunk 4]
```diff
@@ -74,6 +74,97 @@ for ( i = 0; i < 100; i++ ) {
...
+
+```c
+#include "stdlib/math/base/assert/is_probability.h"
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    bool v;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/README.md:149]: Don't we need `stdbool.h` for `bool` macro support?

File: lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    bool v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( ( double ) rand() / ( double ) RAND_MAX ) * 2.0 ) - 1.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 2.0 ) - 1.0;
```

For casts, by convention, we don't surround the type with spaces.

File: lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_probability.h"
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c:20]:
```suggestion
#include <stdio.h>
#include <stdbool.h>
```

File: lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_probability.h"
+#include <math.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c]: What is this header used for?

File: lib/node_modules/@stdlib/math/base/assert/is-probability/README.md [hunk 8]
```diff
@@ -74,6 +74,97 @@ for ( i = 0; i < 100; i++ ) {
...
+
+<!-- C API usage examples. -->
+
+<section class="examples">
+
+### Examples
+
+```c
+#include "stdlib/math/base/assert/is_probability.h"
+#include <math.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/README.md]: What is this header used for?

File: lib/node_modules/@stdlib/math/base/assert/is-probability/src/addon.c [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+		return NULL;
+	}
+
+	double x;
+	status = napi_get_value_double( env, argv[ 0 ], &x );
+	assert( status == napi_ok );
+
+	bool result = stdlib_base_is_probability( x );
+    
+	int32_t out = (int32_t) result;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/src/addon.c]:
```suggestion
	int32_t out = (int32_t)result;
```

File: lib/node_modules/@stdlib/math/base/assert/is-probability/src/addon.c [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+	napi_value v;
+	status = napi_create_int32( env, out, &v );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/src/addon.c]:
```suggestion
	napi_value v;
	status = napi_create_int32( env, (int32_t)result, &v );
```

The additional variable is not necessary.

File: lib/node_modules/@stdlib/math/base/assert/is-probability/src/main.c [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+#include "stdlib/math/base/assert/is_probability.h"
+
+/**
+* Tests if a numeric value is a probability.
+*
+* @param x    input value
+* @return	  output value
+*
+* @example
+* double out = stdlib_base_is_probability( 0.5 );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/src/main.c]:
```suggestion
* #include <stdbool.h>
*
* bool out = stdlib_base_is_probability( 0.5 );
```

File: lib/node_modules/@stdlib/math/base/assert/is-probability/README.md [hunk 12]
```diff
@@ -74,6 +74,97 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/README.md:143]:
```suggestion
#include <stdio.h>
#include <stdbool.h>
```

**Review by kgryte — CHANGES_REQUESTED** [review]
Looking good, @Pranavchiku. Just need to make a few minor changes and clarify which headers are actually needed.

File: lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c (hunk 7 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/assert/is-probability/examples/c/example.c]: I thought `RAND_MAX` needs this, but it does not, we can remove this.

File: lib/node_modules/@stdlib/math/base/assert/is-probability/README.md (hunk 4 above)
Comment (Pranavchiku) [reply lib/node_modules/@stdlib/math/base/assert/is-probability/README.md:149]: Yeah! added it.

**Comment by Pranavchiku** [comment]
All tests cleared, this PR can have a review.

File: lib/node_modules/@stdlib/math/base/assert/is-probability/README.md [hunk 13]
```diff
@@ -74,6 +74,97 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdbool.h>
+
+int main() {
+    double x;
+    bool v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( ( double ) rand() / ( double ) RAND_MAX ) * 2.0 ) - 1.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/README.md]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 2.0 ) - 1.0;
```

File: lib/node_modules/@stdlib/math/base/assert/is-probability/src/main.c [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/assert/is_probability.h"
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/assert/is-probability/src/main.c:21]:
```suggestion
#include <stdbool.h>

```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**